libr/core/cmd_anal.c

```diff
@@ -2779,6 +2780,21 @@ static int cmd_anal_fcn(RCore *core, const char *input) {
 			r_core_cmdf (core, "\"afs %s\"", data);
 			free (sig);
 			free (data);
+		} else if (input[2] == 'r') { // "afsr"
+			if (input[3] == ' ') {
+				const char *ntype = ((input[4]) != '\0')
+					? (input + 4)
+					: NULL;
+				if (ntype) {
+					char *sig = r_core_cmd_str (core, "afs");
+					char *sig_notype = strchr (sig, ' ');
```

Comment:
i think that the way to go would be to find '(' and go backward until theres'a space which will be the function name, then you can trim down, but again , using the api you can skip all this text parsing